libraries/lwIP_enc28j60/src/utility/enc28j60.cpp

```diff
@@ -742,3 +766,106 @@ ENC28J60::readFrameData(uint8_t *buffer, uint16_t framesize)
 
     return _len;
 }
+
+/*---------------------------------------------------------------------------*/
+uint16_t
+ENC28J60::phyRead(uint8_t address)
+{
+    unsigned int            timeout = 15;
+
+    setregbank(REG_BANK_2);
+    
+    // set the PHY register address
+    writereg(MIREGADR, address);
+    setregbitfield(MICMD, 0x01);                // MII Read Enable [MIIRD]
+    
+    setregbank(REG_BANK_3);
+	
+    // wait until the PHY read completes
+    while(readreg(MISTAT) & 0x01){
+        delay(10);
+        wdt_reset();
+        if(--timeout ==0)       return 0;
+    }
+    
+    setregbank(REG_BANK_2);
+    
+    clearregbitfield(MICMD, 0x01);              // MII Read Disable [MIIRD]
+    
+    // read the PHY data
+    return (readreg(MIRDL) | readreg(MIRDH) << 8);
+}
+
+void
+ENC28J60::phyWrite(uint8_t address, uint16_t data)
+{
+    unsigned int            timeout = 15;
+    
+    setregbank(REG_BANK_2);
+    
+    // set the PHY register address
+    writereg(MIREGADR, address);
+    
+    // write the PHY data
+    writereg(MIWRL, data & 0xff);
+    writereg(MIWRH, data >> 8);
+    
+    setregbank(REG_BANK_3);
+    
+    // wait until the PHY write completes
+    while(readreg(MISTAT) & 0x01){
+        delay(10);
+        wdt_reset();
+        if(--timeout ==0)       return;
+    }
+}
+
+//--------------------------------------------------------
+void
+ENC28J60::init(void)
+{
+    pinMode(_cs, OUTPUT);
+    digitalWrite(_cs, HIGH);
+    SPI.begin();
+}
+
+void
+ENC28J60::LEDsConfig(LEDScfg_CONFIG LEDAcfg, LEDScfg_CONFIG LEDBcfg)
+{
+    uint16_t                temp  = phyRead(PHLCON);
+    
+    if(LEDAcfg)             temp &= LEDA_MASK;
+    if(LEDBcfg)             temp &= LEDB_MASK;
+	
+    phyWrite(PHLCON, (temp |(uint16_t)(LEDAcfg <<4) |LEDBcfg |0x3006));
+}
+
+//--------------------------------------------------------
+bool
+ENC28J60::resetEther(void)
```

Comment:
Why is it necessary ?

Reply:
Sometimes the ENC28J60 does not work properly (freezes) and needs to be reconfigured

Reply:
Ah you need to call it from "userland".
Instead of duping it, why don't you move it to the public section ?